libs/langgraph/langgraph/pregel/main.py

```diff
@@ -2246,9 +2262,16 @@ async def aperform_superstep(
                     raise InvalidUpdateError(f"Node {as_node} has no writers")
                 writes: deque[tuple[str, Any]] = deque()
                 task = PregelTaskWrites((), as_node, writes, [INTERRUPT])
+                # get the task ids that were prepared for this node
+                # if a task id was provided in the StateUpdate, we use it
+                # otherwise, we use the next available task id
+                prepared_task_ids = node_to_task_ids.get(as_node, [])
                 task_id = provided_task_id or str(
-                    uuid5(UUID(checkpoint["id"]), INTERRUPT)
```

Comment:
_nit_: might be more readable if instead of a counter we'd popped the task id whenever it has been consumed